### PR TITLE
adds description, ingestion architecture and testing steps in README

### PR DESCRIPTION
**Pull request by JBAhire**
This PR,
- Adds description about Hypertrace-view-generators.
- Adds ingestion pipeline architecture.
- Adds how to run unit tests
- Adds how to test locally built image with docker-compose

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,10 +4,48 @@
...
-It is a bootstrap job that creates required views in pinot like spanEventView, backendEntityView, etc.
+## Description
 
-Hypertrace view generator: 
-It is a streaming job that materializes enriched traces into pinot views
+| ![space-1.jpg](https://hypertrace-docs.s3.amazonaws.com/ingestion-pipeline.png) | 
+|:--:| 
+| *Hypertrace Ingestion Pipeline* |
+
+Hypertrace view creator is a bootstrap job that runs ones and created required views in pinot which involves `spanEventView`, `serviceCallView`, `backendEntityView`, `rawServiceView`, `rawTraceView`.
```
Comment (avinashkolluru) [inline README.md:15]:
```suggestion
Hypertrace view creator is a bootstrap job that runs once and creates required views in pinot -  `spanEventView`, `serviceCallView`, `backendEntityView`, `rawServiceView`, `rawTraceView`.
```

File: README.md [hunk 2]
```diff
@@ -4,10 +4,48 @@
...
+
+Ones this views are there in pinot, Hypertrace view generator takes enriched traces which [Hypertrace trace enricher](https://github.com/hypertrace/hypertrace-trace-enricher) pushed to kafka and materializes them into pinot views.
```
Comment (avinashkolluru) [inline README.md:17]:
```suggestion
Once these views are there in pinot, Hypertrace view generator consumes(from kafka) enriched traces produced by [Hypertrace trace enricher](https://github.com/hypertrace/hypertrace-trace-enricher), applies required filters, transformations and then materializes them into pinot views. 
```

File: README.md [hunk 3]
```diff
@@ -4,10 +4,48 @@
...
-It is a streaming job that materializes enriched traces into pinot views
+| ![space-1.jpg](https://hypertrace-docs.s3.amazonaws.com/ingestion-pipeline.png) | 
+|:--:| 
+| *Hypertrace Ingestion Pipeline* |
+
+Hypertrace view creator is a bootstrap job that runs ones and created required views in pinot which involves `spanEventView`, `serviceCallView`, `backendEntityView`, `rawServiceView`, `rawTraceView`.
+
+Ones this views are there in pinot, Hypertrace view generator takes enriched traces which [Hypertrace trace enricher](https://github.com/hypertrace/hypertrace-trace-enricher) pushed to kafka and materializes them into pinot views. 
+
+For example, Let's say we got a trace with 50 attributes After enrichement by Hypertrace trace enricher and `spanEventView` only needs 10 attributes so Hypertrace view generator will fill that view with only those 10 attributes.
```
Comment (avinashkolluru) [inline README.md:19]: Based on the above statement may be this example is redundant.
```suggestion
```

File: README.md [hunk 4]
```diff
@@ -4,10 +4,48 @@
...
+### Running unit tests
+Run `./gradlew test` to execute unit tests. 
+
+
+### Testing image
+
+You can test the image you built after modification by running docker-compose or helm setup. 
+
+#### docker-compose
+Change the tag for `hypertrace-view-generatorr ` from `:main` to `:test` in [docker-compose file](https://github.com/hypertrace/hypertrace/blob/main/docker/docker-compose.yml) like this.
```
Comment (avinashkolluru) [inline README.md:39]:
```suggestion
Change the tag for `hypertrace-view-generator ` from `:main` to `:test` in [docker-compose file](https://github.com/hypertrace/hypertrace/blob/main/docker/docker-compose.yml) like this.
```

**Review by tim-mwangi — APPROVED** [review]

**Pull request merged**